## Performance por país e segmento

### 🎯Objetivo dessa secção

Analisar como países e segmentos contribuem para a receita e lucro, identificando:

- concentração de resultados
- eficiência (vendas x lucros)
- possíveis distorções de rentabilidade

As principais perguntas para essa atividade são:

✅ Quais países geram **mais receita e lucro**?  
✅ Existe país com **alto volume de vendas, mas baixo lucro**?  
✅ Qual segmento de clientes é mais rentável?  
⬜ O governo compra mais, mas dá mais ou menos lucro que o setor privado?  

In [1]:
import pandas as pd

In [2]:
finances = pd.read_excel('../../database/financial.xlsx')
finances.head()

segment  country    product discount_band  units_sold  \
0      Government   Canada    Montana          High         200   
1  Small Business  Germany  Carretera           Low         214   
2  Small Business  Germany        VTT           Low         214   
3       Midmarket   Canada      Paseo           Low         218   
4      Government  Germany      Paseo          High         241   

   manufacturing_price  sale_price  gross_sales  discounts    sales     cogs  \
0                    5         350      70000.0     9800.0  60200.0  52000.0   
1                    3         300      64200.0     1284.0  62916.0  53500.0   
2                  250         300      64200.0     1284.0  62916.0  53500.0   
3                   10          15       3270.0      130.8   3139.2   2180.0   
4                   10          20       4820.0      482.0   4338.0   2410.0   

   profit       date  month_number month_name  year  
0  8200.0 2014-05-01             5        May  2014  
1  9416.0 2013-10-01            10    October  2013  
2  9416.0 2013-10-01            10    October  2013  
3   959.2 2014-09-01             9  September  2014  
4  1928.0 2014-10-01            10    October  2014

In [8]:
# Receita e Lucro por país
country_performance = (
    finances
        .groupby('country')[['sales', 'profit']]
        .sum()
        .reset_index()
        .sort_values('sales', ascending=False)
)

country_performance

country        sales      profit
4  United States of America  25029830.18  2995540.68
0                    Canada  24887654.89  3529228.89
1                    France  24354172.29  3781020.79
2                   Germany  23505340.82  3680388.82
3                    Mexico  20949352.11  2907523.11

In [ ]:
# Existe país com alto volume de vendas, mas baixo lucro?

# Obter relatório de vendas e lucro por país, para o calculo de margem de lucro
country_margin = (
    finances
        .groupby('country')[['sales', 'profit']]
        .sum()
        .reset_index()
)

# Calcular a margem de lucro através das colunas Lucro (profit) e Vendas (sales)
country_margin['profit_margin'] = (
    country_margin['profit'] / country_margin['sales']
)

# Ordenar Países com as maiores vendas e verificar se a margem foi mais baixa que os restantes.
country_margin.sort_values('sales', ascending=False)

country        sales      profit  profit_margin
4  United States of America  25029830.18  2995540.68       0.119679
0                    Canada  24887654.89  3529228.89       0.141806
1                    France  24354172.29  3781020.79       0.155251
2                   Germany  23505340.82  3680388.82       0.156577
3                    Mexico  20949352.11  2907523.11       0.138788

In [ ]:
# Qual segmento de clientes é mais rentável?

# Obter relatório de qual segmento é mais rentável para o calculo de margem de lucro por segmento.
segment_performance = (
    finances
        .groupby('segment')[['sales', 'profit']]
        .sum()
        .reset_index()
        .sort_values('sales', ascending=False)
)

# Calculo de margem de lucro por segmento
segment_performance['profit_margin'] = (
    segment_performance['profit'] / segment_performance['sales']
)

# Ordenar resultados por lucro e margem de lucro para uma melhor interpretação dos dados.
(segment_performance
 .sort_values(['profit', 'profit_margin'], ascending=[False, False]))

segment        sales       profit  profit_margin
2        Government  52504260.68  11388173.18       0.216900
4    Small Business  42427918.50   4143168.50       0.097652
0  Channel Partners   1800593.64   1316803.14       0.731316
3         Midmarket   2381883.09    660103.09       0.277135
1        Enterprise  19611694.38   -614545.62      -0.031336

In [ ]:
# O governo compra mais, mas dá mais ou menos lucro que o setor privado?

segment_comparison = (
    finances
        .groupby("segment")[["sales", "profit"]]
        .sum()
        .reset_index()
)

segment_comparison["profit_margin"] = (
    segment_comparison["profit"] / segment_comparison["sales"]
)

segment_comparison.sort_values(['sales', 'profit'], ascending=[False, False])


segment        sales       profit  profit_margin
2        Government  52504260.68  11388173.18       0.216900
4    Small Business  42427918.50   4143168.50       0.097652
1        Enterprise  19611694.38   -614545.62      -0.031336
3         Midmarket   2381883.09    660103.09       0.277135
0  Channel Partners   1800593.64   1316803.14       0.731316

## 🔎 Insights – Performance por País e Segmento

### 🌍 Receita e Lucro por País
A análise por país mostra uma distribuição relativamente equilibrada de faturamento entre **Estados Unidos, Canadá, França e Alemanha**, todos com volumes superiores a 23 milhões.

Entretanto, **França e Alemanha** apresentam maior lucro absoluto, mesmo sem liderarem em faturamento, indicando maior eficiência operacional.

---

### ⚖️ Alto Volume de Vendas x Baixo Lucro
Os **Estados Unidos** lideram em faturamento, porém apresentam a **menor margem de lucro (~12%)**, sugerindo maior pressão de custos ou política de descontos mais agressiva.

---

### 🧑‍💼 Rentabilidade por Segmento
O segmento **Government** se destaca como o mais relevante financeiramente, combinando alto faturamento e **margem elevada (~21,7%)**.

O segmento **Small Business** apresenta margem reduzida (~9,8%), enquanto o segmento **Enterprise** registra **prejuízo**, mesmo com volume relevante de vendas.

---

### 🏛️ Governo x Setor Privado
Os dados indicam que o **segmento governamental é altamente rentável**, contrariando a percepção comum de margens comprimidas nesse tipo de contrato.

---

### 🧠 Considerações Finais da Etapa
- Maior faturamento não garante maior rentabilidade  
- Governo é o principal motor de lucro  
- Segmentos e países com baixo retorno exigem revisão estratégica  